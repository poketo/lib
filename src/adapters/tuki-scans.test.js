import poketo from '../index';
import site from './tuki-scans';

describe('TukiScansAdapter', () => {
  const server = new AdapterVcrServer(site);

  beforeAll(async () => {
    await server.listenAndMock(57170);
  });

  afterAll(() => {
    server.close();
  });

  describe('getSeries', () => {
    it('returns a metadata object', async () => {
      const { chapters, ...metadata } = await poketo.getSeries(
        'tuki-scans:madoromi-chan',
      );

      expect(metadata).toMatchSnapshot();

      const chapterNumbersToTest = ['17'];
      const chaptersToTest = chapters.filter(chapter =>
        chapterNumbersToTest.includes(chapter.chapterNumber),
      );
      expect(chaptersToTest).toMatchSnapshot();
    });
  });
});
