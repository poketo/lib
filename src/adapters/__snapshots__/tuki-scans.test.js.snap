// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TukiScansAdapter getSeries returns a metadata object 1`] = `
Object {
  "coverImageUrl": "https://reader.tukimoop.pw/content/comics/madoromi-chan_5a5435fb56565/thumb_thumb.jpg",
  "slug": "madoromi-chan",
  "title": "Madoromi-chan ga Iku.",
  "url": "http://localhost:57170/series/madoromi-chan",
}
`;

exports[`TukiScansAdapter getSeries returns a metadata object 2`] = `
Array [
  Object {
    "chapterNumber": "17",
    "createdAt": 1526129849,
    "slug": "en/2/17",
    "url": "http://localhost:57170/read/madoromi-chan/en/2/17/page/1",
    "volumeNumber": "2",
  },
]
`;
